doc/architecture/adr/0021-PROPOSAL-health-checks.md

```diff
@@ -0,0 +1,39 @@
+# Health chekcs
+
+* Status: Proposed
+* Deciders:
+* Date: 2023-05-07
+
+---
+
+## Context and Problem Statement
+
+We are building multiple services and we must be able to react if a service is not responding or is experiencing performance issues. Furthermore, changes in performed in the infrastructure (e.g. updating or refactoring configuration) can cause services to break. We must have a simple way of ensuring our changes performed in the infrastructure does not effect a service.
+
+---
+
+## Considered Options
+
+* Health checks
+
+A health check is a process that periodically checks the status of a service by sending a request and checking for a response. A service that responds with a status code within a certain range (e.g., 200-299) is considered healthy, while a service that does not respond or responds with an error status code is considered unhealthy.
+
+---
+
+## Decision Outcome
+
+We will use health checks to monitor the availability and health of our services.
+
+We will use a health check endpoint provided by the .NET Core Health Checks middleware. This middleware provides a simple way to define health checks for our services. The endpoint can be accessed via an HTTP GET request to a specified URL. We can define custom health checks to test specific aspects of our services, such as database connectivity, valid configuration, external service dependencies or application-specific features.
+
```

Comment:
```suggestion

We will ensure our health checks covers all non-external dependencies before becoming healthy.

```

Reply:
What is a non-external dependency vs. an external dependency?

Reply:
We should check provided configuration, required files, connections to databases/caches/queues, etc. and not check if a OIDC provider is healthy nor if another domain is healthy.

I am open to use better wording to express this, if you have a suggestion :)